engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -74,14 +78,20 @@ class NextgenEngine @JvmOverloads constructor(nadel: Nadel, transforms: List<Nad
             executionInput.variables,
         )
 
+        val serviceContexts = services.values.map {
```

Comment:
Also, this is the default way that you create a `Map` in Kotlin, but we don't really like this approach.

Here it's simple and fine, but if you have entries with duplicate keys, then they just get merged together. This ends up often times masking an underlying error.

Instead you can use:

```
mapFrom(
  service.values.map {
    it.name to …
  }
)
```

Which ends up throwing an error if the size of input entries != size of the output map. We could probably name it better, not sure.